app/schema.graphql

```diff
@@ -0,0 +1,103 @@
+scalar Datetime
+
+type Author {
+  name: String
+  email: String
+}
+
+# git commit data.
+type Commit {
+  sha1: String!
+  author: Author!
+  date: Datetime!
+  message: String!
+  refs: [Ref]
+  objectName: ObjectName!
+}
+
+# file info and metadata.
+type Info {
+  # base name of the file.
+  name: String!
+  # length in bytes.
+  size: Int!
+  # if it's a directory.
+  isdirectory: Boolean!
+  # any flags git stores, currently only executable.
+  flags: [Int]!
+  # last commit which affected this path.
+  lastcommit: Commit!
+}
+
+type BlobEntry {
+  content: String!
+  info: Info!
+}
+
+type TreeEntry {
+  path: String!
+  info: Info!
+}
+
+# Representation of a refname. Special refs like HEAD are omitted on purpose as
+# of now.
+#
+# master
+# v1.0.0
+# heads/master
+# tags/v1.0.0
+# remotes/origin/master
+type Ref = {
```

Comment:
I think I need a better story for refs vs branches vs tags. At the moment I just refs for _everything_ and haven't tested it with a commit SHA1 to see if that works.

Reply:
We used to have `union Ref = Head | Remote | Tag`. But when you ignore special refs like `HEAD` and friends this type works quite well. Was pretty much working along this: https://git-scm.com/book/en/v2/Git-Internals-Git-References

Reply:
Cool! I'll take a look :)